### PR TITLE
Position-sensitive ROI Pool/Align

### PR DESCRIPTION
**Pull request by sampepose**
This adds PS ROI pool and align. I also fixed a couple bugs with existing ROI pool tests.

### CODE REVIEW

**Comment by LukasBommes** [comment]
I'm already using the layer and it works fine!

**Comment by fmassa** [comment]
I will get this reviewed and merged into torchvision soon.
But before we need to have CI for windows and OSX working, so that we can make sure that everything is working fine on the OS that we support, see https://github.com/pytorch/vision/pull/1298

File: torchvision/csrc/PSROIPool.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const int pooled_width) {
+  if (input.type().is_cuda()) {
+#ifdef WITH_CUDA
+    return PSROIPool_forward_cuda(
+        input, rois, spatial_scale, pooled_height, pooled_width);
+#else
+    AT_ERROR("Not compiled with GPU support");
+#endif
+  }
+  AT_ERROR("Not implemented on the CPU");
```
Comment (fmassa) [inline torchvision/csrc/PSROIPool.h]: torchvision follows similar requirements as PyTorch with respect to the implementation of the ops: it requires the ops to be implemented for both CPU and GPU.

**Comment by LukasBommes** [comment]
@sampepose Are you working on this? Or should I give it a try?

**Comment by LukasBommes** [comment]
I'm currently working on the CPU implementation as I rely on this layer for another project.

**Comment by fmassa** [comment]
This has been subsumed by #1410. Thanks for the original version @sampepose!

**Pull request closed without merging**